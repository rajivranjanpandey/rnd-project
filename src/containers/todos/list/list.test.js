import { fireEvent, render, screen, within } from "@testing-library/react"
import TodoList from "."
const props = {
    todos: [
        {
            id: 1,
            title: 'Learn React'
        },
        {
            id: 2,
            title: 'Learn Javascript'
        }
    ],
    onDeleteTodoItem: jest.fn()
}
describe('TodoList', () => {
    it('Check todolist rendering', () => {
        const { asFragment } = render(<TodoList {...props} />)
        expect(asFragment(<TodoList />)).toMatchSnapshot();
    });
    it('Check deletion of todolist', () => {
        render(<TodoList {...props} />);
        const mappedTodos = screen.getAllByTestId('todo-item').map((item) => ({
            title: within(item)
                .getByTestId('title')
                .textContent,
            deleteButton: within(item)
                .getByRole('button')
        }));
        const deleteElement = mappedTodos.find(item => item.title === 'Learn React').deleteButton;
        fireEvent.click(deleteElement);
        expect(props.onDeleteTodoItem).toBeCalled();
    })
})