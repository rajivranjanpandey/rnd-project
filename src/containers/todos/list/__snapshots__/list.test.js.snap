// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Check todolist rendering 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="css-1k9ek97"
    >
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
          data-testid="todo-item"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="TurnedInIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            data-testid="title"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Learn React
            </span>
          </div>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              aria-label="delete"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
          data-testid="todo-item"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="TurnedInIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            data-testid="title"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Learn Javascript
            </span>
          </div>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              aria-label="delete"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
